exchanges/bybit/bybit_wrapper.go

```diff
@@ -2418,13 +1895,67 @@ func getContractLength(contractLength time.Duration) (futures.ContractType, erro
 	return ct, nil
 }
 
-// IsPerpetualFutureCurrency ensures a given asset and currency is a perpetual future
-func (by *Bybit) IsPerpetualFutureCurrency(a asset.Item, p currency.Pair) (bool, error) {
-	return a == asset.Futures && p.Quote.Equal(currency.PFC), nil
-}
-
 // GetLatestFundingRates returns the latest funding rates data
-func (by *Bybit) GetLatestFundingRates(context.Context, *fundingrate.LatestRateRequest) ([]fundingrate.LatestRateResponse, error) {
-	// TODO: implement with v5 API upgrade
-	return nil, common.ErrNotYetImplemented
+func (by *Bybit) GetLatestFundingRates(ctx context.Context, r *fundingrate.LatestRateRequest) ([]fundingrate.LatestRateResponse, error) {
```

Comment:
Tickers contain the current funding rates and next time period. Diff for implementation. 
```diff
diff --git a/exchanges/bybit/bybit_wrapper.go b/exchanges/bybit/bybit_wrapper.go
index 2f580d89e..b120640a3 100644
--- a/exchanges/bybit/bybit_wrapper.go
+++ b/exchanges/bybit/bybit_wrapper.go
@@ -1903,54 +1903,47 @@ func (by *Bybit) GetLatestFundingRates(ctx context.Context, r *fundingrate.Lates
        if r.IncludePredictedRate {
                return nil, fmt.Errorf("%w IncludePredictedRate", common.ErrFunctionNotSupported)
        }
-       var err error
-       if r.Pair.IsEmpty() {
-               return nil, currency.ErrCurrencyPairEmpty
-       }
-       var format currency.PairFormat
-       format, err = by.GetPairFormat(r.Asset, true)
-       if err != nil {
-               return nil, err
-       }
-       r.Pair = r.Pair.Format(format)

        switch r.Asset {
        case asset.USDCMarginedFutures,
                asset.USDTMarginedFutures,
                asset.CoinMarginedFutures:


-               var fRates *FundingRateHistory
-               fRates, err = by.GetFundingRateHistory(ctx, getCategoryName(r.Asset), r.Pair.String(), time.Time{}, time.Time{}, 0)
+               symbol := ""
+               if !r.Pair.IsEmpty() {
+                       format, err := by.GetPairFormat(r.Asset, true)
+                       if err != nil {
+                               return nil, err
+                       }
+                       symbol = r.Pair.Format(format).String()
+               }
+
+               ticks, err := by.GetTickers(ctx, getCategoryName(r.Asset), symbol, "", time.Time{})
                if err != nil {
                        return nil, err
                }

-               var latestTimestamp int64
-               resp := make([]fundingrate.LatestRateResponse, 1)
-               for i := range fRates.List {
+               resp := make([]fundingrate.LatestRateResponse, 0, len(ticks.List))
+               for i := range ticks.List {
                        var cp currency.Pair
                        var isEnabled bool
-                       cp, isEnabled, err = by.MatchSymbolCheckEnabled(fRates.List[i].Symbol, r.Asset, true)
+                       cp, isEnabled, err = by.MatchSymbolCheckEnabled(ticks.List[i].Symbol, r.Asset, false)
                        if err != nil && !errors.Is(err, currency.ErrPairNotFound) {
                                return nil, err
-                       } else if !isEnabled ||
-                               !cp.Equal(r.Pair) ||
-
-                               // check if FundingRateTimestamp is earlier than latestTimestamp.
-                               fRates.List[i].FundingRateTimestamp.Time().UnixMilli() <= latestTimestamp {
+                       } else if !isEnabled {
                                continue
                        }
-                       latestTimestamp = fRates.List[i].FundingRateTimestamp.Time().UnixMilli()
-                       resp[0] = fundingrate.LatestRateResponse{
+
+                       resp = append(resp, fundingrate.LatestRateResponse{
                                TimeChecked: time.Now(),
                                Exchange:    by.Name,
                                Asset:       r.Asset,
                                Pair:        cp,
                                LatestRate: fundingrate.Rate{
-                                       Time: fRates.List[i].FundingRateTimestamp.Time(),
-                                       Rate: decimal.NewFromFloat(fRates.List[i].FundingRate.Float64()),
+                                       Rate: decimal.NewFromFloat(ticks.List[i].FundingRate.Float64()),
                                },
-                       }
+                               TimeOfNextRate: time.UnixMilli(ticks.List[i].NextFundingTime.Int64()),
+                       })
                }
                if len(resp) == 0 {
                        return nil, fmt.Errorf("%w %v %v", futures.ErrNotPerpetualFuture, r.Asset, r.Pair)
```

Reply:
Please keep the `LatestRate: fundingrate.Rate.Time` though

Reply:
You can also include the predicted rate `predictedFundingFeeRate` field if `if r.IncludePredictedRate {`

Reply:
The diff doesn't apply. I think you used a different documentation or code branch.
However, I have added the Next Funding time information.

Reply:
@samuael You need to go through and add those changes, they were brought acrost from a different branch I am integrating and using. The current implementation is not the latest rate and it's slow. Please see above and make those changes using the above as a guide.

Reply:
I have applied the change above. Thank you for the detailed explanation.

Reply:
Thanks Sam. Please add the fields I mention above